Extensions/ParticleSystem/particleemitterobject.js

```diff
@@ -1,56 +1,169 @@
 /**
-
-GDevelop - Particle System Extension
-Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
-This project is released under the MIT License.
+ * @fileoverview
+ * @name Particle-System-Extension
+ * @copyright Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
+ * @author Florian Rival
+ * @license MIT
+ * GDevelop - Particle System Extension
+ * Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
+ * This project is released under the MIT License.
 */
 
 
-gdjs.ParticleEmitterObject = function(runtimeScene, objectData){
-    gdjs.RuntimeObject.call(this, runtimeScene, objectData);
-
-    this._renderer = new gdjs.ParticleEmitterObjectRenderer(runtimeScene, this, objectData);
-
-    this.singleAngle = objectData.emissionEditionSimpleMode;
-    this.angleA = objectData.emitterAngleA;
-    this.angleB = objectData.emitterAngleB;
-    this.forceMin = objectData.emitterForceMin;
-    this.forceMax = objectData.emitterForceMax;
-    this.zoneRadius = objectData.zoneRadius;
-    this.lifeTimeMin = objectData.particleLifeTimeMin;
-    this.lifeTimeMax = objectData.particleLifeTimeMax;
-    this.gravityX = objectData.particleGravityX;
-    this.gravityY = objectData.particleGravityY;
-    this.colorR1 = objectData.particleRed1;
-    this.colorR2 = objectData.particleRed2;
-    this.colorG1 = objectData.particleGreen1;
-    this.colorG2 = objectData.particleGreen2;
-    this.colorB1 = objectData.particleBlue1;
-    this.colorB2 = objectData.particleBlue2;
-    this.size1 = objectData.particleSize1;
-    this.size2 = objectData.particleSize2;
-    this.sizeParam = objectData.sizeParam;
-    this.alpha1 = objectData.particleAlpha1;
-    this.alpha2 = objectData.particleAlpha2;
-    this.rendererType = objectData.rendererType;
-    this.rendererParam1 = objectData.rendererParam1;
-    this.rendererParam2 = objectData.rendererParam2;
-    /** @type string */
-    this.texture = objectData.textureParticleName;
-    this.flow = objectData.flow;
-    this.tank = objectData.tank;
-    this.destroyWhenNoParticles = objectData.destroyWhenNoParticles;
+// FIX-ME @4ian correct with real types (thoose are assunmptions only) and add desciptions
```

Comment:
Please @4ian I am not sure about the types can you check?

Reply:
Thank you!

Reply:
emissionEditionSimpleMode is a boolean
rendererType is a string

The rest looks correct :)

Reply:
You fixed that I think? should this be marked as resolved?